httpbakery/error_test.go

```diff
@@ -0,0 +1,70 @@
+package httpbakery_test
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+
+	gc "gopkg.in/check.v1"
+
+	"gopkg.in/macaroon-bakery.v0/httpbakery"
+	"gopkg.in/macaroon.v1"
+)
+
+type ErrorSuite struct{}
+
+var _ = gc.Suite(&ErrorSuite{})
+
+func (s *ErrorSuite) TestWriteDischargeRequiredError(c *gc.C) {
+	m, err := macaroon.New([]byte("secret"), "id", "a location")
+	c.Assert(err, gc.IsNil)
+	tests := []struct {
+		about        string
+		path         string
+		err          error
+		expectedBody string
+	}{
+		{
+			`write discharge required with "an error" but no path`,
+			"",
+			errors.New("an error"),
+			fmt.Sprintf(`{"Code":"macaroon discharge required","Message":"an error","Info":{"Macaroon":{"caveats":[],"location":"%s","identifier":"%s","signature":"%v"}}}`,
```

Comment:
I suggest importing juju/httptesting and using AssertJSONResponse - then you can write these
as simple Error literals.